clients/gtest/her_gtest.cpp

```diff
@@ -0,0 +1,139 @@
+/* ************************************************************************
+ * Copyright 2018-2020 Advanced Micro Devices, Inc.
+ *
+ * ************************************************************************ */
+
+#include "rocblas_data.hpp"
+#include "rocblas_datatype2string.hpp"
+#include "rocblas_test.hpp"
+#include "testing_her.hpp"
+#include "testing_her_batched.hpp"
+#include "testing_her_strided_batched.hpp"
+#include "type_dispatch.hpp"
+#include <cctype>
+#include <cstring>
+#include <type_traits>
+
+namespace
+{
+    // possible test cases
+    enum her_test_type
+    {
+        HER,
+        HER_BATCHED,
+        HER_STRIDED_BATCHED,
+    };
+
+    //her test template
+    template <template <typename...> class FILTER, her_test_type HER_TYPE>
+    struct her_template : RocBLAS_Test<her_template<FILTER, HER_TYPE>, FILTER>
+    {
+        // Filter for which types apply to this suite
+        static bool type_filter(const Arguments& arg)
+        {
+            return rocblas_simple_dispatch<her_template::template type_filter_functor>(arg);
+        }
+
+        // Filter for which functions apply to this suite
+        static bool function_filter(const Arguments& arg)
+        {
+            switch(HER_TYPE)
+            {
+            case HER:
+                return !strcmp(arg.function, "her") || !strcmp(arg.function, "her_bad_arg");
+            case HER_BATCHED:
+                return !strcmp(arg.function, "her_batched")
+                       || !strcmp(arg.function, "her_batched_bad_arg");
+            case HER_STRIDED_BATCHED:
+                return !strcmp(arg.function, "her_strided_batched")
+                       || !strcmp(arg.function, "her_strided_batched_bad_arg");
+            }
+            return false;
+        }
+
+        // Google Test name suffix based on parameters
+        static std::string name_suffix(const Arguments& arg)
+        {
+            RocBLAS_TestName<her_template> name;
+
+            name << rocblas_datatype2string(arg.a_type);
+
+            if(strstr(arg.function, "_bad_arg") != nullptr)
+            {
+                name << "_bad_arg";
+            }
+            else
+            {
+                // For alpha: T doesn't really matter here, just whether it's real or complex, in her's case it's always complex.
+                name << '_' << (char)std::toupper(arg.uplo) << '_' << arg.N << '_' << arg.lda << '_'
+                     << arg.get_alpha<rocblas_double_complex>() << '_' << arg.incx;
```

Comment:
This is her so it's always real not complex right?   THis looks like it is her2 code where it is always complex.